### PR TITLE
implementing statement

### PR DESCRIPTION
**Pull request by andrefurlan-db**
Basic implementation of statement.

There is no support in databricks for transactions or query parameters. So for now we are returning not supported errors.

Signed-off-by: Andre Furlan <andre.furlan@databricks.com>

### CODE REVIEW

File: connection.go [hunk 1]
```diff
@@ -63,7 +76,14 @@ func (c *conn) IsValid() bool {
...
 
+// ExecContext executes a query that doesn't return rows, such
+// as an INSERT or UPDATE.
+//
+// ExecContext honors the context timeout and return when it is canceled.
 func (c *conn) ExecContext(ctx context.Context, query string, args []driver.NamedValue) (driver.Result, error) {
+	if len(args) > 0 {
+		return nil, fmt.Errorf("databricks: query parameters are not supported")
+	}
 	_, opStatusResp, err := c.runQuery(ctx, query, args)
```
Comment (mattdeekay) [inline connection.go:87]: Is there a difference between `cli_service.TExecuteStatementResp.GetModifiedRowCount()` vs `cli_service.TGetOperationStatusResp.GetNumModifiedRows()`? I was using the former in my implementation

File: connection.go [hunk 2]
```diff
@@ -118,26 +145,26 @@ func (c *conn) runQuery(ctx context.Context, query string, args []driver.NamedVa
...
 				return nil, statusResp, err
 			}
 			switch statusResp.GetOperationState() {
 			// terminal states
 			// good
-			case ts.TOperationState_FINISHED_STATE:
+			case cli_service.TOperationState_FINISHED_STATE:
 				// return handle to fetch results later
 				return exStmtResp, opStatus, nil
 			// bad
```
Comment (mattdeekay) [inline connection.go:170]: Nit: a lot of nesting of cases here, no functional problems but I'll pull this out

File: connection.go (hunk 1 above)
Comment (andrefurlan-db) [reply connection.go:87]: haha, good question. Can you check and submit a PR if this needs to be changed?

File: connection.go (hunk 2 above)
Comment (andrefurlan-db) [reply connection.go:170]: yes. But it is still pretty readable, so I left the way it is to not complicate further. If it bothers you feel free to make the changes!

**Review by mattdeekay — APPROVED** [review]

**Review by rcypher-databricks — APPROVED** [review]

**Pull request merged**